src/v/cluster/security_frontend.cc

```diff
@@ -339,4 +342,83 @@ security_frontend::get_bootstrap_user_creds_from_env() {
       std::in_place, std::move(username), std::move(credentials));
 }
 
+ss::future<result<model::offset>> security_frontend::get_leader_committed(
+  model::timeout_clock::duration timeout) {
+    auto leader = _leaders.local().get_leader(model::controller_ntp);
+    if (!leader) {
+        return ss::make_ready_future<result<model::offset>>(
+          make_error_code(errc::no_leader_controller));
+    }
+
+    if (leader == _self) {
+        return ss::smp::submit_to(controller_stm_shard, [this, timeout]() {
+            return ss::with_timeout(
+                     model::timeout_clock::now() + timeout,
+                     _controller->linearizable_barrier())
+              .handle_exception_type([](const ss::timed_out_error&) {
+                  return result<model::offset>(errc::timeout);
+              });
+        });
+    }
+
+    return _connections.local()
+      .with_node_client<cluster::controller_client_protocol>(
+        _self,
+        ss::this_shard_id(),
+        *leader,
+        timeout,
+        [timeout](controller_client_protocol cp) mutable {
+            return cp.get_controller_committed_offset(
+              controller_committed_offset_request{},
+              rpc::client_opts(model::timeout_clock::now() + timeout));
+        })
+      .then(&rpc::get_ctx_data<controller_committed_offset_reply>)
+      .then([](result<controller_committed_offset_reply> r) {
+          if (r.has_error()) {
+              return result<model::offset>(r.error());
+          } else if (r.value().result != errc::success) {
+              return result<model::offset>(r.value().result);
+          }
+          return result<model::offset>(r.value().last_committed);
+      });
+}
+
+ss::future<std::error_code> security_frontend::wait_until_caughtup_with_leader(
```

Comment:
Why is this a security frontend interface? This seems like it should be generic.

Reply:
I didn't want to offer this as a generic solution where there are currently no other consumers, figured someone else could break it out if they needed. I'm fine with either FWIW

Reply:
>figured someone else could break it out if they needed

how will the find it?